### PR TITLE
fix OcTree.castRay to set end if success

### LINKED ISSUE #3: cythoning error
Hi

I attempted to install the package and get an error when creating octomap.cpp

`Error compiling Cython file:
------------------------------------------------------------
...
        else:
            raise NullPointerException

    def dynamicEDT_update(self, updateRealDist):
        if self.edtptr:
            return self.edtptr.update(<cppbool?>updateRealDist)
                                    ^
------------------------------------------------------------

octomap/octomap.pyx:813:37: Cannot convert 'void' to Python object
`
I deleted the function dynamicEDT_update, and the everything built nicely. Any ideas what is going on? I am not versed in cython.

Thanks,
Rohan

### LINKED ISSUE #5: OcTree.castRay never sets end
even when `OcTree.castRay` return `True` the `end=` values are never changed

### PR DESCRIPTION
**Pull request by daizr**
and added `test_castRay` too.

closes #3, closes #5

### CODE REVIEW

File: octomap/octomap.pyx [hunk 1]
```diff
@@ -810,7 +818,7 @@ cdef class OcTree:
 
     def dynamicEDT_update(self, updateRealDist):
         if self.edtptr:
-            return self.edtptr.update(<cppbool?>updateRealDist)
+            self.edtptr.update(<cppbool?>updateRealDist)
```
Comment (daizr) [inline octomap/octomap.pyx]: def looks like:

```cython
cdef extern from "dynamicEDT3D/dynamicEDTOctomap.h":
    cdef cppclass DynamicEDTOctomap:
        ...
        void update(bool updateRealDist)
```

so i ran into error about cannot convert `void` to `PyObject` so i just rm return

File: test/octomap_test.py [hunk 2]
```diff
@@ -88,5 +88,41 @@ def test_Iterator(self):
...
+
+        # hit
+        hit = self.tree.castRay(
+            origin=origin,
+            direction=direction,
+            end=end,
+            ignoreUnknownCells=True,
+        )
+        self.assertTrue(hit)
+        self.assertTrue(np.allclose(end, [1.05, 0.05, 0.05]))
```
Comment (daizr) [inline test/octomap_test.py]: before this patch always `[0, 0, 0]`

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+octomap/octomap.cpp
```
Comment (daizr) [inline .gitignore]: seems cython generates this file `octomap/octomap.cpp` so i assime we want to ignore?

**Comment by neka-nat** [comment]
Thank you for fixing bug and adding the test!!

**Pull request merged**